### PR TITLE
Remove usages of Z3_TRUE / Z3_FALSE.

### PR DESCRIPTION
**Pull request by waywardmonkeys**
Now that this is all using stdbool.h, we can just use true/false.

For now, we leave the aliases in place in z3_api.h.

### CODE REVIEW

File: src/api/api_arith.cpp [hunk 1]
```diff
@@ -121,7 +121,7 @@ extern "C" {
 
     Z3_bool Z3_API Z3_is_algebraic_number(Z3_context c, Z3_ast a) {
         LOG_Z3_is_algebraic_number(c, a);
-        return mk_c(c)->autil().is_irrational_algebraic_numeral(to_expr(a)) ? Z3_TRUE : Z3_FALSE;
+        return mk_c(c)->autil().is_irrational_algebraic_numeral(to_expr(a)) ? true : false;
```
Comment (NikolajBjorner) [inline src/api/api_arith.cpp]: if we are extra courageous, we may even do

      return mk_c(c)->autil().is_irrational_algebraic_numeral(to_expr(a))

File: src/api/api_numeral.cpp [hunk 2]
```diff
@@ -188,7 +188,7 @@ extern "C" {
         CHECK_IS_EXPR(a, "");
         rational r;
         Z3_bool ok = Z3_get_numeral_rational(c, a, r);
-        if (ok == Z3_TRUE) {
+        if (ok == true) {
```
Comment (NikolajBjorner) [inline src/api/api_numeral.cpp]: if (ok) works too

File: src/api/api_seq.cpp [hunk 3]
```diff
@@ -70,35 +70,35 @@ extern "C" {
...
     }
 
     Z3_bool Z3_API Z3_is_re_sort(Z3_context c, Z3_sort s) {
         Z3_TRY;
         LOG_Z3_is_re_sort(c, s);
         RESET_ERROR_CODE();
         bool result = mk_c(c)->sutil().is_re(to_sort(s));
-        return result?Z3_TRUE:Z3_FALSE;
-        Z3_CATCH_RETURN(Z3_FALSE);
+        return result?true:false;
```
Comment (NikolajBjorner) [inline src/api/api_seq.cpp]: another few levels of indirection.

**Comment by waywardmonkeys** [comment]
I'll submit another PR within a day or two that cleans up those as well.

**Comment by NikolajBjorner** [comment]
I went over what I could find.

**Comment by waywardmonkeys** [comment]
You got all of the ones that I saw! Thanks.

**Pull request merged**